frontend/src/pages/Konva.js

```diff
@@ -0,0 +1,158 @@
+        const y = e.target.y();
+
+        console.log(
+            'ID: ' + id + '  ' +
+            'X: ' + x + '  ' +
+            'Y: ' + y)
+    }
+
+    const setRectCoordinatesOnDragEnd = (e) => {
+        const id = e.target.id();
+        const x = e.target.x();
+        const y = e.target.y();
+
+        setRects(
+            rects.map((rect) => {
+                if (rect.id === id) {
+                    return {
+                        ...rect,
+                        x: x,
+                        y: y,
+                        isDragging: false,
+                        width: 30,
+                        height: 30,
+                        shadowBlur: 10,
+                        fill: '#C9E1F2', shadowOffsetX: 5, shadowOffsetY: 5
+                    }
+                } else {
+                    return {...rect}
+                }
+            }))
+
+        console.log(
+            'ID: ' + id + '  ' +
+            'X: ' + x + '  ' +
+            'Y: ' + y)
+    }
+
+    const setCircleCoordinatesOnDragEnd = (e) => {
+        const id = e.target.id();
+        const x = e.target.x();
+        const y = e.target.y();
+
+        setCircles(
+            circles.map((circle) => {
+                if (circle.id === id) {
+                    return {
+                        ...circle,
+                        x: x,
+                        y: y,
+                    }
+                } else {
+                    return {...circle}
+                }
+            }))
+        
+        console.log(
+            'ID: ' + id + '  ' +
+            'X: ' + x + '  ' +
+            'Y: ' + y)
+    }
+
+    const logRects = () => {
+        console.log(rects)
+    }
+
+    const logCircles = () => {
+        console.log(circles)
+    }
+
+    return (
+        <>
+            <button onClick={addRect}>Add Rect</button>
+            <button onClick={addCircle}>Add Circle</button>
+            <button onClick={logRects}>Log Rects</button>
+            <button onClick={logCircles}>Log Circles</button>
+            <Stage width={window.innerWidth} height={window.innerHeight} background={'#fff'}>
+                <Layer>
+                    {rects.map((rect) => (
+                        <Rect
+                            key={rect.id}
+                            id={rect.id}
+                            x={rect.x}
+                            y={rect.y}
+                            width={rect.width}
+                            height={rect.height}
+                            fill={rect.fill}
+                            stroke={rect.stroke}
+                            strokeWidth={rect.strokeWidth}
+                            shadowBlur={rect.shadowBlur}
+                            draggable
+                            isDragging={false}
+                            onDragStart={handleDragStart}
+                            onDragEnd={setRectCoordinatesOnDragEnd}
+                        />))}
```

Comment:
```suggestion
                        <Rect
                            {...rect}
                            key={rect.id}
                            draggable
                            isDragging={false}
                            onDragStart={handleDragStart}
                            onDragEnd={setRectCoordinatesOnDragEnd}
                        />))}
```
spread operator is your friend ;)